### PR TITLE
API modifications and cleanup

### PR DESCRIPTION
**Pull request by adamdbrw**
- Moved some classes that are useful for development of robotic projects to API.
- Cleaned up MotorizedJoint documentation, naming.
- Renamed MotorizedJoint to MotorizedJointComponent

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

### CODE REVIEW

**Comment by adamdbrw** [comment]
With reference to https://github.com/RobotecAI/o3de-ros2-gem/issues/313.

File: Code/Include/ROS2/Manipulator/MotorizedJointComponent.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#include "ROS2/VehicleDynamics/DriveModels/PidConfiguration.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector2.h>
+
+namespace ROS2
+{
+    //! A prototype component for simulated joint with a motor.
+    //! It works with either TransformBus and RigidBodyBus.
```
Comment (michalpelka) [inline Code/Include/ROS2/Manipulator/MotorizedJointComponent.h]:
```suggestion
    //! It works with either TransformBus or RigidBodyBus.
```

File: Code/Include/ROS2/Manipulator/MotorizedJointComponent.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private:
+        float ComputeMeasurement(AZ::ScriptTimePoint time);
+        void SetVelocity(float velocity, float deltaTime);
+        void ApplyLinVelAnimation(float velocity, float deltaTime);
+        void ApplyLinVelRigidBodyImpulse(float velocity, float deltaTime);
+        void ApplyLinVelRigidBody(float velocity, float deltaTime);
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        AZ::Vector3 m_jointDir{ 0.f, 0.f, 1.f }; //!< Direction of joint movement in parent frame of reference, used to compute measurement.
+        AZ::Vector3 m_effortAxis{ 0.f, 0.f, 1.f }; //!< Direction of force application in owning entity frame of reference.
```
Comment (michalpelka) [inline Code/Include/ROS2/Manipulator/MotorizedJointComponent.h]:
```suggestion
        AZ::Vector3 m_effortAxis{ 0.f, 0.f, 1.f }; //!< Direction of force or torque application in owning entity frame of reference.
````

File: Code/Source/Manipulator/MotorizedJointComponent.cpp [hunk 3]
```diff
@@ -25,85 +25,95 @@ namespace ROS2
...
-                ec->Class<MotorizedJoint>("MotorizedJoint", "MotorizedJoint")
-                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "MotorizedJoint")
+                ec->Class<MotorizedJointComponent>("MotorizedJointComponent", "MotorizedJointComponent")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "MotorizedJointComponent")
                     ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
-                    ->Attribute(AZ::Edit::Attributes::Category, "MotorizedJoint")
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_jointDir, "Joint Dir.", "Direction of joint.")
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_effortAxis, "Effort Dir.", "Direction of joint.")
+                    ->Attribute(AZ::Edit::Attributes::Category, "MotorizedJointComponent")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_jointDir, "Joint Dir.", "Direction of joint.")
```
Comment (michalpelka) [inline Code/Source/Manipulator/MotorizedJointComponent.cpp]:
```suggestion
                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_jointDir, "Joint Dir.", "Direction of joint in parent's reference frame.")
````

File: Code/Source/Manipulator/MotorizedJointComponent.cpp [hunk 4]
```diff
@@ -25,85 +25,95 @@ namespace ROS2
...
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_effortAxis, "Effort Dir.", "Direction of joint.")
```
Comment (michalpelka) [inline Code/Source/Manipulator/MotorizedJointComponent.cpp]:
```suggestion
                        AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_effortAxis, "Effort Dir.", "Desired direction of force/torque vector that is applied to rigid body.")
```

**Review by michalpelka — APPROVED** [review]
Looks good, some suggestions from my site.

**Pull request merged**